### PR TITLE
Add package for progress bars and test program to display several bars and a spinner

### PR DESCRIPTION
**Pull request by onox**

### CODE REVIEW

File: src/progress_indicators-bars.adb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+--  SPDX-License-Identifier: Apache-2.0
+--
+--  Copyright (c) 2021 onox <denkpadje@gmail.com>
```
Comment (pyjarrett) [inline src/progress_indicators-bars.adb]: The standard here should be:

```
Copyright (c) 2021 The progress_indicators authors
```

and then do an CONTRIBUTORS.TXT file at the repo root.

File: src/progress_indicators-bars.adb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+with Ada.Strings.Fixed;
+
+package body Progress_Indicators.Bars is
+
+    function Get_Bar (Value : Percentage; Width : Natural := 100) return String is
+        use Ada.Strings.Fixed;
+        use ANSI;
+
+        Used : constant Natural := Natural (Float (Width) / 100.0 * Float (Value));
+        Remaining : constant Natural := Width - Used;
```
Comment (pyjarrett) [inline src/progress_indicators-bars.adb:28]: We should have aligned colons here.

File: src/progress_indicators-bars.adb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+--  See the License for the specific language governing permissions and
+--  limitations under the License.
+
+with ANSI;
+
+with Ada.Strings.Fixed;
+
+package body Progress_Indicators.Bars is
+
+    function Get_Bar (Value : Percentage; Width : Natural := 100) return String is
```
Comment (pyjarrett) [inline src/progress_indicators-bars.adb:23]: I had originally been thinking of something more persistent, but I definitely like this stateless idea better.

File: src/progress_indicators-bars.adb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        use Ada.Strings.Fixed;
+        use ANSI;
+
+        Used : constant Natural := Natural (Float (Width) / 100.0 * Float (Value));
+        Remaining : constant Natural := Width - Used;
+
+        Text : constant String := Tail (Value'Image & "%", 5);
+    begin
+        return
+          Foreground (Light_Green) & Used * "█" & Foreground (Default) & Remaining * "░" & Text &
```
Comment (pyjarrett) [inline src/progress_indicators-bars.adb:32]: I think this is fine for now, but we're probably going to eventually need to be able to make ANSI a no-op for systems that don't have it, by wrapping the calls or something.

File: src/progress_indicators-bars.adb (hunk 3 above)
Comment (onox) [reply src/progress_indicators-bars.adb:23]: Yeah, it doesn't really need any state.

File: src/progress_indicators-bars.adb (hunk 1 above)
Comment (onox) [reply src/progress_indicators-bars.adb]: I've added a CONTRIBUTORS.md file so that it gets rendered by GitHub.

**Review by pyjarrett — APPROVED** [review]

**Pull request merged**